File: Intent/MyIntentApp/java/app/src/main/java/com/dicoding/picodiploma/myintentapp/MoveWithObjectActivity.java
package com.dicoding.picodiploma.myintentapp;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class MoveWithObjectActivity extends AppCompatActivity {
    public static final String EXTRA_PERSON = "extra_person";
    private TextView tvObject;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_move_with_object);

        tvObject = findViewById(R.id.tv_object_received);

        /*
        Object parcelable bisa kita dapatkan dengan memanggil getParcelableExtra
         */
        Person person = getIntent().getParcelableExtra(EXTRA_PERSON);

        String text = "Name : " + person.getName() + ",\nEmail : " + person.getEmail() + ",\nAge : " + person.getAge() + ",\nLocation : " + person.getCity();
        tvObject.setText(text);
    }
}

File: Intent/MyIntentApp/kotlin/app/src/main/java/com/dicoding/picodiploma/myintentapp/MoveWithObjectActivity.kt
package com.dicoding.picodiploma.myintentapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_move_with_object.*

class MoveWithObjectActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_PERSON = "extra_person"
    }
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_move_with_object)

        /*
        Object parcelable bisa kita dapatkan dengan memanggil getParcelableExtra
         */
        val person = intent.getParcelableExtra<Person>(EXTRA_PERSON)

        val text = "Name : ${person.name},\nEmail : ${person.email},\nAge : ${person.age},\nLocation : ${person.city}"
        tv_object_received.text = text
    }


}
